xmppserver/src/main/java/org/jivesoftware/openfire/session/LocalSession.java

```diff
@@ -502,11 +503,16 @@ public Certificate[] getPeerCertificates() {
                 .orElse(new Certificate[0]);
     }
 
+    // TODO: Remove this override. The override (and the boolean property) serves as a emergency fallback for the change in OF-3031 and _should not_ be needed. It should be desirable to use #getStatus() only.
     @Override
     public boolean isClosed() {
-        return Optional.ofNullable(conn)
+        if (JiveGlobals.getBooleanProperty("xmpp.session.isclose.connectionbased", false)) {
+            return Optional.ofNullable(conn)
                 .map(Connection::isClosed)
                 .orElse(Boolean.TRUE);
+        } else {
+            return getStatus() == Status.CLOSED;
```

Comment:
super.isClosed(), perhaps?

Reply:
I don't think that works with default methods on interfaces.